control_panel_api/services.py

```diff
@@ -153,6 +153,17 @@ def delete_bucket_policies(bucket_name):
     aws.delete_policy(policy_arn_readonly)
 
 
+def apps3bucket_create(apps3bucket):
+    app_slug = apps3bucket.app.slug
+    bucket_name = apps3bucket.s3bucket.name
+    readwrite = (apps3bucket.access_level == READWRITE)
```

Comment:
This is a code smell - the function requires a lot of knowledge of the internals of the apps3bucket object - this should probably be a method on the class.

Reply:
Originally this function was receiving these arguments as a string.

I know this is usually a symptom of methods being in the wrong place.
At the same time, don't we risk to overload models with a lot of methods? In the past I've worked on some applications in which we had models for basic crud stuff, and service objects to add a layer of context-aware operations - As in methods used when performing certain operations in certain contexts, or maybe even validations only happening in certain context, e.g. admin interface could have different validations or something like that.

I feel like this comment is a more general one on the general architecture and structure of the app rather than something specific of this feature changes itself: Should we "ignore" this for the specific feature and have an offline discussion with @jmoz about of we want to structure things and refactor existing stuff?

Reply:
@andyhd and @jmoz to recap what just discussed offline:
- We're going to refactor and move these functions in classes.
- We'll merge these PRs and we'll move them as a separate step/PRs
- We'll look at the functions and see if there is some natural grouping which would lead to ideas on where to put this code (potentially start by moving them in the models and maybe further down the line move them in "proper" service classes)